### ISSUE TITLE
levelinfo fog color does not save

### ISSUE THREAD

**Issue by ghost** [GarageGames/Torque3D#1043]
If you change the color of the fog in the levelinfo object and you close and reload the level the fog color is reset to the default. The color data you chose is still stored in the datablock, but it does not get loaded when you load the level, only if you open the editor and touch the value again.

**Comment by John3** [comment]
Hi @Duion,

What version are you using when the issue is present? Can you shared a bit more of information?

I tested this in the actual development version without problem. I can change the fogcolor and canvascolor, save and restart engine, the change persist.

<img src="http://i.imgur.com/FTZQWWY.png"/>

Can we consider close this? unless the bug still exist. @Areloch

**Comment by Areloch** [comment]
Thanks for reminding me about this, John.

Did a fast trace-through of what's going on: turns out ScatterSky superscedes the levelinfo fog's setting on startup so that you can control the fog and whatnot with the scattersky object. If you're using a skybox, that doesn't occur, which is why you weren't seeing it in your test.

If you use the default empty terrain, you'll see that it sets to whatever the scattersky's fog color is set to.

This is a "working as intended" thing, not a bug, but I'll agree that it can be confusing disorienting to have it set-able in two places, depending on what sky you have in the level and whatnot. I figure we can either leave this ticket open for discussion on a better way to implement it, or likely more preferabbly make a new ticket to discuss a better approach to handling fog in a level.

**Comment by Duion** [comment]
Yes, I already found that out quite a while ago, that it is supersceded by the scattersky, but did not know that I posted this here as an issue.
So this is not really a bug, it is just confusing, if you don't know where to set it.
